.github/workflows/my-ci-pipeline.yml

```diff
@@ -41,3 +41,32 @@ jobs:
           SNYK_TOKEN: ${{ secrets.SNYK_TOKEN }}
         with:
           command: monitor
+
+  deployment:
+    name: Deploy to production
+    runs-on: ubuntu-latest
+    needs: [build]
+    if: github.ref == 'refs/heads/main'
+    steps:
+      - run: echo Publishing!
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
```

Comment:
Very minor but slightly interesting - you've put your username as a secret here, but then you include it below as part of the Docker tag (`talexandru87/todo-app:prod`). Interestingly, this leads GitHub to actually mask/hide your username in the tag, meaning that anyone who has access to this file and the CI logs (both of which are public!) can absolutely confirm that the secret is exactly that string below. Not a big deal because fundamentally this _isn't_  a secret here, but an interesting side-effect. E.g. if you used a common word as a password, like `password` - then your CI logs might actually betray that fact elsewhere!